### PR TITLE
[hackage] switch to the `/licenses` endpoint

### LINKED ISSUE #1498: hackage-deps: incorrect `not found` check
`hackage-deps` claims to check existence of package (say `3d-graphics-examples`)  by looking for packages that depend on it. If there are none then `not found` badge is returned, instead of checking the dependencies of `3d-graphics-examples`.

### PR DESCRIPTION
**Pull request by chris48s**
Closes #1498

Switch to the `/licenses` endpoint to see if the package exists on the `hackage-deps` badge.

Unrelated: one of the other hackage tests failed due to some trailing whitespace on a version number, so I've also added a fix for that.

### CODE REVIEW

File: server.js [hunk 1]
```diff
@@ -3293,6 +3293,7 @@ cache(function(data, match, sendBadge, request) {
       // we'll render the 'invalid' badge below, which is the correct thing
       // to do.
       var version = versionLines[0].replace(/\s+/, '').split(/:/)[1];
+      version = version.trim();
```
Comment (RedSparr0w) [inline server.js]: What do you think if we just use something like:
```javascript
var version = versionLines[0].split(/:/)[1].trim();
```

Comment (chris48s) [reply server.js]: that's neater - yes :)

**Comment by RedSparr0w** [comment]
Thanks!
Changes work well.
Merged 😄

**Comment by jakubdaniel** [comment]
> Changes work well.
Really? [https://img.shields.io/hackage-deps/v/3d-graphics-examples.svg](https://img.shields.io/hackage-deps/v/3d-graphics-examples.svg) still gives me `not found`.

**Comment by chris48s** [comment]
Although this fix has been merged to master, having a quick look at the deploy branch, it has not yet been deployed to production.

**Pull request merged**